src/plugins/eui_utils/public/eui_chart_utils.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreStart, CoreSetup } from 'src/core/public';
+import { EUI_CHARTS_THEME_DARK, EUI_CHARTS_THEME_LIGHT } from '@elastic/eui/dist/eui_charts_theme';
+
+export class EuiChartUtils {
+  public setup(core: CoreSetup) {}
+  public start(core: CoreStart) {
+    return {
+      getChartsTheme() {
+        return core.uiSettings.get('theme:darkMode')
```

Comment:
One thing that would be nice is if in the future we could allow live toggling of the theme without having to refresh the entire browser. We could have dashboard-specific themes and what not.

If we made this usage here an Observable, we'd make it easier to build that feature in the future. In general, we want the UI system to be adaptive to things changing at any time.

To do that here, we could do this instead:

```diff
   public setup(core: CoreSetup) {}
   public start(core: CoreStart) {
     return {
-      getChartsTheme() {
-        return core.uiSettings.get('theme:darkMode')
+      getChartsTheme$() {
+        return core.uiSettings.get$('theme:darkMode')
           ? EUI_CHARTS_THEME_DARK.theme
           : EUI_CHARTS_THEME_LIGHT.theme;
```

Then in each consuming component, you could use the `useEffect()` hook so that all UI gets updated via React state updates.

```js
import React, { useEffect } from 'react';
import { EuiChartThemeType } from '@elastic/eui/src/themes/charts/themes';
import { EUI_CHARTS_THEME_LIGHT } from '@elastic/eui/dist/eui_charts_theme';

const MyCompenent: FunctionComponent = ({ chartsTheme$: Observable<EuiChartThemeType> }) => {
  const [chartsTheme, setChartsTheme] = useState<EuiChartThemeType>(EUI_CHARTS_THEME_LIGHT);
  useEffect(() => {
    const subscription = chartsTheme$.subscribe(setChartsTheme);
    return subscription.unsubscribe();
  }, [chartsTheme$]);

  return (
    <Chart size={['100%', 300]} renderer="canvas">
      <Settings
        theme={chartsTheme}
        {...rest of chart...}
      </Settings>
    </Chart>
  );
};
```

Now we have another feature that gets closer to be able to do on the fly theming and a more adaptive UI. This also helps reinforce this as a pattern, just like the Core APIs do.

Reply:
I do remember that we have a convenience hook for subscriptions (`use_observable`) in `kibana_react`, but it looks like it isn't currently exposed to the other plugins. You could export that one too and use it here to make this component even simpler, but either way works.

Reply:
Great, let me check that it still all works correctly if I change it to the observable.

> I do remember that we have a convenience hook for subscriptions (use_observable) in kibana_react, ... You could export that one too ...

It's a bit over my head to make that kind of a call, so it's whatever you think is best, I can attempt it.

Reply:
Yeah, I'm having all sorts of issues trying to get it setup the way you have it above. Even just `chartsTheme$: Observable<EuiChartThemeType>` is giving me a hard time saying `'EuiChartThemeType' only refers to a type, but is being used as a value here.`

Could I lean on you some more to help get the Discover chart working then I can try to replicate it in the Watcher chart? Or we could put this part off for a bit? It's just quite out of my league.

Reply:
Sure thing, I can take care of it and push it up.

Reply:
Thank you 🎉 💯

Reply:
Just pushed up!

Reply:
Pushed up those changes!